File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int low = 0;
        int sum = 0;
        int minlength = Integer.MAX_VALUE;
        int n = nums.length;
        for(int high = 0;high<n;high++){
            sum += nums[high];
            while(sum>=target){
                sum -= nums[low];
                
                int currentlength = high-low+1;
                minlength = Math.min(minlength,currentlength);
                low++;
            }
            
        }
        if(minlength==Integer.MAX_VALUE){
            return 0;
        }
        return minlength;
        
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        left = 0
        sum = 0
        minlength = float("inf")

        for right in range(len(nums)):
            sum += nums[right]

            while sum >= target:
                currentlength = right - left + 1
                minlength = min(minlength, currentlength)
                sum -= nums[left]
                left += 1

        
        return 0 if minlength == float('inf') else minlength